FrameworkSystem/DB/ProxyDB.py

```diff
@@ -600,18 +682,87 @@ def __getPUSProxy(self, userDN, userGroup, requiredLifetime, requestedVOMSAttr=N
       return S_ERROR('Requested DN does not match the obtained one in the PUSP proxy')
     timeLeft = credDict['secondsLeft']
 
-    result = chain.generateProxyToString(lifeTime=timeLeft,
-                                         diracGroup=userGroup)
+    result = chain.generateProxyToString(timeLeft, diracGroup=userGroup)
     if not result['OK']:
       return result
     proxyString = result['Value']
     return S_OK((proxyString, timeLeft))
 
+  def __generateProxyFromProxyProvider(self, userDN, proxyProvider):
+    """ Get proxy from proxy provider
+
+        :param basestring userDN: user DN for what need to create proxy
+        :param basestring proxyProvider: proxy provider name that will ganarete proxy
+
+        :return: S_OK(dict)/S_ERROR() -- dict with remaining seconds, proxy as a string and as a chain
+    """
+    if not proxyProvider:
```

Comment:
the error is not really that you did not find a proxy provider, but more that none was given. This is a bit misleading no ?

Reply:
error message moved to another method